daffodil-core/src/main/scala/org/apache/daffodil/core/dsom/ElementBase.scala

```diff
@@ -256,15 +256,16 @@ trait ElementBase
 
       // Creates a Map[prefix, Set[NS]]. Duplicate NS's will be removed from the
       // Set, since it's a Set
-      val bindingsGroupedByPrefix = allBindings.groupBy { _._1 }.mapValues { _.map { _._2 } }
+      val bindingsGroupedByPrefix =
```

Comment:
You do not have to rewrite all these mapValues. I believe you can just change `.mapValues` to `.view.mapValues`. This gives a `MapView`, which behaves like a map, except you have to call `toMap` at the end of the chain of operations at some point, possibly, unless a `MapView` will do the job entirely. I think `MapView` is not serializable. 

But I imagine you considered that already and decided to avoid the MapView thing. If so, can you explain why?

Reply:
view.mapValues.toMap didn't work because it breaks 2.12 with the below error. .map(...k -> vMapped) seems like the best alternative

value mapValues is not a member of scala.collection.IterableView[(String, scala.collection.immutable.Set[(String, org.apache.daffodil.lib.xml.NS)]),scala.collection.immutable.Map[String,scala.collection.immutable.Set[(String, org.apache.daffodil.lib.xml.NS)]]]
[error]         allBindings.groupBy { _._1 }.view.mapValues { _.map { _._2 } }.toMap
[error]                                           ^